### DISCUSSION TITLE
Next.js & Module Federation

### DISCUSSION THREAD

**Discussion by wityan** [vercel/next.js#20395] [Help]
Is there a nice solution to use Next.js with module federation? By the word nice I don't mean the Monorepo approach everyone is currently making tutorials with :)

Thx for your Help

**Comment by estevaolucas** [comment]
You probably can refer to this repo: https://github.com/module-federation/nextjs-mf and https://github.com/module-federation/module-federation-examples

There are some challenges to implement MF in a Nex.js app. Some here: https://github.com/module-federation/module-federation-examples/tree/master/nextjs#challenges (and I'm facing others)

**Comment by wityan** [comment]
@estevaolucas yes exactly. I think the Monorepo approach here is a bad idea, but I wasn't able to find a better solution by now.
